src/test/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupportSpec.scala

```diff
@@ -13,105 +13,178 @@ class DataUseRestrictionSupportSpec extends FreeSpec with Matchers with DataUseR
+          val attrs: Map[AttributeName, Attribute] = generateStructuredUseRestrictionAttribute(workspace)
+          attrs should be(empty)
         }
+
+        "dataset should not have any data use restriction for non-library attributes" in {
+          val nonLibraryAttributes = Map(
+            AttributeName.withDefaultNS("name") -> AttributeString("one"),
+            AttributeName.withDefaultNS("namespace") -> AttributeString("two"),
+            AttributeName.withDefaultNS("workspaceId") -> AttributeString("three"),
+            AttributeName.withDefaultNS("authorizationDomain") -> AttributeValueList(Seq(AttributeString("one"), AttributeString("two"), AttributeString("three")))
+          )
+          val workspace: Workspace = mkWorkspace(nonLibraryAttributes, "non-library", "non-library")
+          val attrs: Map[AttributeName, Attribute] = generateStructuredUseRestrictionAttribute(workspace)
+          attrs should be(empty)
+        }
+
       }
 
     }
 
-    "when there are no library data use restriction fields" - {
+    "Display Use Restriction" - {
 
-      "dataset should not have any data use restriction for empty attributes" in {
-        val workspace = mkWorkspace(Map.empty[AttributeName, Attribute], "empty", "empty")
-        val attrs = generateStructuredUseRestriction(workspace)
-        attrs should be(empty)
+      "when there are library data use restriction fields" - {
+
+        "valid datasets should have some form of data use display attribute" in {
+          validDisplayDatasets.map { ds =>
+            val attrs: Map[AttributeName, Attribute] = generateUseRestrictionDisplayAttribute(ds)
+            val codes: Seq[String] = getValuesFromAttributeValueListAsAttribute(attrs.get(dataUseDisplayAttributeName))
+            codes shouldNot be(empty)
+          }
+        }
+
+        "datasets with single boolean code should have that single display code" in {
+          booleanDatasets.map { ds =>
+            val attrs: Map[AttributeName, Attribute] = generateUseRestrictionDisplayAttribute(ds)
+            val codes: Seq[String] = getValuesFromAttributeValueListAsAttribute(attrs.get(dataUseDisplayAttributeName))
+            // Boolean datasets are named with the same code value
+            codes should contain(ds.name)
+            codes.size should be(1)
+          }
+        }
+
+        "'EVERYTHING' dataset should have the right codes" in {
+          val attrs: Map[AttributeName, Attribute] = generateUseRestrictionDisplayAttribute(everythingDataset.head)
+          val codes: Seq[String] = getValuesFromAttributeValueListAsAttribute(attrs.get(dataUseDisplayAttributeName))
+          (booleanCodes ++ Seq("RS-G", "RS-FM", "NAGR")).map { c => codes should contain(c) }
+          diseaseValuesLabels.map { c => codes should contain("DS:" + c) }
+          Seq("RS-M", "RS-POP").map { c => codes shouldNot contain(c) }
+        }
```

Comment:
I generally don't like using `contain` tests, because it's too easy to miss a value on the left or right. Could you instead create the set of expected codes, and compare equality to (or `theSameElementsAs`) the actual codes?